INFRA-417 Improve driver DSL test stability

rnicoll opened the pull request:
This improves the driver DSL test stability by having the tests wait for the nodes to start before connecting to the node. This is likely to be further improvements we can make, but this improves the state for a relatively small set of changes.

Commit:
INFRA-417 Move debug log messages to debug level

Move log messages that are not useful in typical usage from info to debug level to reduce log spam.

Commit:
INFRA-417 Add node startup check

Add node startup check before attempting to connect.

Commit:
Remove unused change

Commit:
Correct future used to determine node is ready

Commit:
Remove unused change

Commit:
Clean up polling usage

adagys commented on testing/node-driver/src/main/kotlin/net/corda/testing/driver/Driver.kt:
```diff
@@ -247,7 +248,7 @@ data class DriverParameters(
         val driverDirectory: Path = Paths.get("build") / "node-driver" / getTimestampAsDirectoryName(),
         val portAllocation: PortAllocation = incrementalPortAllocation(),
-        val debugPortAllocation: PortAllocation = incrementalPortAllocation(),
+        val debugPortAllocation: PortAllocation = portAllocation, // Share port allocator by default
```
Technically this change isn't needed as `incrementalPortAllocation()` will always return the same allocator instance. I raised a related bug, this port allocation config should be cleaned up as now we only really have one allocation strategy and there's nothing to configure.

rnicoll replied:
Removed `debugPortAllocation`, figure broader cuts to this should wait for later.

adagys replied:
Can we do this though? It's an API break. Maybe best to leave it for when the general cleanup happens (if ever)

rnicoll replied:
Getting approval from Matthew & Stefano, we'll see how it goes!

adagys commented on testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt:
```diff
@@ -218,16 +219,24 @@ class DriverDSLImpl(
 
-    private fun establishRpc(config: NodeConfig, processDeathFuture: CordaFuture<out Process>): CordaFuture<CordaRPCOps> {
+    /**
+     * @param connectRetries number of times to try reconnecting before giving up.
```
missing parameter?

rnicoll replied:
An experiment I removed but apparently missed that the docs remained. Will fix.

adagys commented on testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt:
```diff
@@ -673,6 +682,7 @@ class DriverDSLImpl(
                     }
             )
             val nodeFuture: CordaFuture<NodeHandle> = nodeAndThreadFuture.flatMap { (node, thread) ->
+                node.node.nodeReadyFuture.get() // Wait for the node to be ready before we connect to the node
```
Hm, I guess this changes semantics a bit, this method is expected to return a future of a starting node. But if there are no immediate issues with other tests, probably ok.

rnicoll replied:
Previously it still didn't return until there was a client connection, so it shouldn't change semantics.

In an ideal world, yes it would return immediately, and the connection would happen when separately (potentially as a process triggered from the node ready future), but we'd need to make wider changes than I want to tackle today.

adagys commented on testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt:
```diff
@@ -779,6 +789,7 @@ class DriverDSLImpl(
     }
 
     companion object {
+        private val DEFAULT_CONNECT_POLL_INTERVAL: Duration = 100.millis
```
why is this needed? there is an interval default in the poll method already

rnicoll replied:
Reducing the poll interval to try encouraging the tests to pass a bit faster. The previous value was 500ms.

adagys commented on testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt:
```diff
@@ -218,16 +219,24 @@ class DriverDSLImpl(
+     */
+    private fun establishRpc(config: NodeConfig,
+                             processDeathFuture: CordaFuture<out Process>,
+                             pollInterval: Duration = DEFAULT_CONNECT_POLL_INTERVAL): CordaFuture<CordaRPCOps> {
```
the new parameter isn't used?

rnicoll replied:
`pollInterval = pollInterval) {` below

adagys replied:
it doesn't seem to be set to anything other than the default value though, and there is already a default value set within the `poll` method

rnicoll replied:
The idea is that the calling code _can_ decide this, and understands that there is a value which defines poll interval, rather than it being defined elsewhere and opaque to the calling code.

Commit:
Remove debug port allocation

Remove debug port allocation as all port allocators are now the same shared allocator under the hood.

Commit:
Remove Javadoc from an unused experiment

roastario reviewed (dismissed):
in _THEORY_ this shouldn't have an impact as all port allocators are now actually just proxies to the global shared-memory one.

Commit:
Update API - to be approved by Matthew

Commit:
Revert controversial changes so we can get this actually merged

Commit:
Remove parameter for poll interval

mnesbit reviewed (approved):
OK
